### PR TITLE
Feature regex

### PR DESCRIPTION
**Pull request by rom10**

### CODE REVIEW

**Comment by rom10** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;Qo7N5R3s&#x2F;34-zendesk-worker"> Zendesk Worker</a>

File: circle.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 machine:
   node:
-    version: 0.10.40
+    version: 4.2.2
```
Comment (asafdav) [inline circle.yml]: This might be an issue as we use this package in meteor

File: lib/handlers/multiHandler.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+util.inherits(MultiMsgHandler, MsgHandler);
+
+/**
+ * Check if the topic supposed to handle in this handler
+ * Using RegEx
+ * @param routingKey
+ */
+MultiMsgHandler.prototype.match = function (routingKey) {
```
Comment (asafdav) [inline lib/handlers/multiHandler.js]: Do we have to create a new msg handler ? 
I think we can just make this the deafult.

File: lib/handlers/multiHandler.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+var MsgHandler = require('./handler');
+
+/**
+ *
+ * @param name
+ * @param pattern
+ * @param topics (optional) - Define the acceptable topics for the handler, could be a list of topics, or a dict
+ * @constructor
+ */
+function MultiMsgHandler(name, pattern, topics) {
```
Comment (asafdav) [inline lib/handlers/multiHandler.js]: Following our conversation, let's merge pattern and topics into one parameter

File: circle.yml (hunk 1 above)
Comment (rom10) [inline circle.yml]: Changed

File: lib/handlers/multiHandler.js (hunk 2 above)
Comment (rom10) [inline lib/handlers/multiHandler.js]: Changed

File: lib/handlers/multiHandler.js (hunk 3 above)
Comment (rom10) [inline lib/handlers/multiHandler.js]: Merged

File: circle.yml (hunk 1 above)
Comment (asafdav) [inline circle.yml]: doesn't seem like it

File: lib/handlers/handler.js [hunk 4]
```diff
@@ -16,6 +16,22 @@ function MsgHandler(name, pattern) {
  * @param routingKey
  */
 MsgHandler.prototype.match = function(routingKey) {
+  // If array, check if the topic is inside the array
+  if (_.isArray(this.pattern)) {
+    return (this.pattern.indexOf(routingKey) != -1);
```
Comment (asafdav) [inline lib/handlers/handler.js]: What if it's an array of regexps ?

**Pull request closed without merging**